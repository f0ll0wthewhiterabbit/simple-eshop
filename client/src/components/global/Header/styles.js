import styled from 'styled-components'
import { IconButton, AppBar, Typography } from '@material-ui/core'

export const Wrapper = styled(AppBar)`
  height: ${props => props.theme.sizing.headerHeight.xs};
  position: relative;
  z-index: 2;

  ${props => props.theme.breakpoints.up('sm')} {
    height: ${props => props.theme.sizing.headerHeight.sm};
  }
`

export const MenuButton = styled(IconButton)`
  color: ${props => props.theme.palette.primary.contrastText};

  &:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }

  ${props => props.theme.breakpoints.up('lg')} {
    display: none;
  }
`

export const WarningMessage = styled(Typography)`
  margin-right: 12px;
`
