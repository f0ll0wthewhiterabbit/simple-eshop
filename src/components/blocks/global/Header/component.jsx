import React from 'react'

const Header = () => {
  return (
    <header>
      Header content should be here...
    </header>
  )
}

export default Header
