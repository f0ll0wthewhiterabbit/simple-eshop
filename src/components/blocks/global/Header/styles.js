import styled from 'styled-components'
import ShopTwoRoundedIcon from '@material-ui/icons/ShopTwoRounded'
import { Typography } from '@material-ui/core'

const LogoWrapper = styled.a`
  display: block;
  flex-grow: 1;
  display: flex;
  align-items: center;
`

const LogoIcon = styled(ShopTwoRoundedIcon)`
  margin-right: ${props => props.theme.spacing(1)}px;
`

const LogoTitle = styled(Typography)`
  display: none;

  ${props => props.theme.breakpoints.up('sm')} {
    display: block;
  }
`
export { LogoIcon, LogoTitle, LogoWrapper }
