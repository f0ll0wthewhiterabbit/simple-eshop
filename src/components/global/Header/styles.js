import styled from 'styled-components'
import { IconButton, AppBar } from '@material-ui/core'

const Wrapper = styled(AppBar)`
  height: ${props => props.theme.sizing.headerHeight.xs};
  position: relative;
  z-index: 2;

  ${props => props.theme.breakpoints.up('sm')} {
    height: ${props => props.theme.sizing.headerHeight.sm};
  }
`

const MenuButton = styled(IconButton)`
  color: ${props => props.theme.palette.primary.contrastText};

  &:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }

  ${props => props.theme.breakpoints.up('lg')} {
    display: none;
  }
`

export { Wrapper, MenuButton }
