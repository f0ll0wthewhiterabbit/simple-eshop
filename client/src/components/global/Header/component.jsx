import React from 'react'
import PropTypes from 'prop-types'
import { Toolbar, Container } from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import Logo from './components/Logo'
import UserMenu from './components/UserMenu'
import { Wrapper, MenuButton, WarningMessage } from './styles'

const Header = ({ openSidebar, isAdmin, isAuthenticated, isDeleteRequestSent }) => {
  const handleMenuButtonClick = () => {
    openSidebar()
  }

  return (
    <Wrapper position="static">
      <Container maxWidth="lg">
        <Toolbar>
          {isAdmin && (
            <MenuButton color="secondary" aria-label="open sidebar" onClick={handleMenuButtonClick}>
              <MenuIcon />
            </MenuButton>
          )}
          <Logo />
          {isAuthenticated && isDeleteRequestSent && (
            <WarningMessage variant="body2" color="error">
              Unfortunately, your account will be deleted soon.
            </WarningMessage>
          )}
          {isAuthenticated && <UserMenu />}
        </Toolbar>
      </Container>
    </Wrapper>
  )
}

Header.propTypes = {
  openSidebar: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  isDeleteRequestSent: PropTypes.bool.isRequired,
}

export default Header
