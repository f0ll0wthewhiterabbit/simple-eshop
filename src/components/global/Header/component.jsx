import React from 'react'
import PropTypes from 'prop-types'
import { Toolbar, Container } from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu'

import Logo from './components/Logo'
import UserMenu from './components/UserMenu'
import { Wrapper, MenuButton } from './styles'

const Header = ({ openSidebar, isAdminMode, isUserSignedUp }) => {
  const handleMenuButtonClick = () => {
    openSidebar()
  }

  return (
    <Wrapper position="static">
      <Container maxWidth="lg">
        <Toolbar>
          {isAdminMode && (
            <MenuButton color="secondary" aria-label="open sidebar" onClick={handleMenuButtonClick}>
              <MenuIcon />
            </MenuButton>
          )}
          <Logo />
          {isUserSignedUp && <UserMenu />}
        </Toolbar>
      </Container>
    </Wrapper>
  )
}

Header.propTypes = {
  openSidebar: PropTypes.func.isRequired,
  isAdminMode: PropTypes.bool.isRequired,
  isUserSignedUp: PropTypes.bool.isRequired,
}

export default Header
