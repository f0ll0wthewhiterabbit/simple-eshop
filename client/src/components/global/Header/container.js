import { connect } from 'react-redux'

import Header from './component'
import { openSidebar } from '../../../store/actions'
import { DATABASE_FIELD_ROLE_ADMIN } from '../../../constants'

const mapStateToProps = state => ({
  isAdmin: state.getIn(['auth', 'user', 'role']) === DATABASE_FIELD_ROLE_ADMIN,
  isAuthenticated: state.getIn(['auth', 'isAuthenticated']),
  isDeleteRequestSent: state.getIn(['auth', 'user', 'isRemovable']),
})

const mapDispatchToProps = dispatch => ({
  openSidebar: () => dispatch(openSidebar()),
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
