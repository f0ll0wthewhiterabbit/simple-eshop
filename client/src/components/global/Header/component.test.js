import React from 'react'
import { shallow } from 'enzyme'

import Header from './component'
import findByTestAttr from '../../../utils/findByTestAttr'

describe('Header component', () => {
  let props
  let wrapper
  const openSidebar = jest.fn()
  const dataTestMenuButton = 'menuButton'
  const dataTestWarningMessage = 'warningMessage'
  const dataTestUserMenu = 'userMenu'

  const generateWrapper = passedProps => {
    const defaultProps = {
      openSidebar,
      isAdmin: false,
      isAuthenticated: true,
      isDeleteRequestSent: false,
    }
    props = { ...defaultProps, ...passedProps }

    return shallow(<Header {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should contain user menu button', () => {
    const button = findByTestAttr(wrapper, dataTestUserMenu)

    expect(button).toHaveLength(1)
  })

  it(`shouldn't contain user menu button if not authenticated`, () => {
    wrapper = generateWrapper({ isAuthenticated: false })
    const button = findByTestAttr(wrapper, dataTestUserMenu)

    expect(button).toHaveLength(0)
  })

  it(`shouldn't contain warning message for users who sent account delete request`, () => {
    wrapper = generateWrapper({ isDeleteRequestSent: true })
    const warningMessage = findByTestAttr(wrapper, dataTestWarningMessage)

    expect(warningMessage).toHaveLength(1)
  })

  it('should contain menu button in admin mode', () => {
    wrapper = generateWrapper({ isAdmin: true })
    const button = findByTestAttr(wrapper, dataTestMenuButton)

    expect(button).toHaveLength(1)
  })

  it(`shouldn't contain menu button if not admin`, () => {
    const button = findByTestAttr(wrapper, dataTestMenuButton)

    expect(button).toHaveLength(0)
  })

  it('should open sidebar after menu button click', () => {
    wrapper = generateWrapper({ isAdmin: true })
    const button = findByTestAttr(wrapper, dataTestMenuButton)
    button.simulate('click')

    expect(openSidebar).toHaveBeenCalledTimes(1)
  })
})
