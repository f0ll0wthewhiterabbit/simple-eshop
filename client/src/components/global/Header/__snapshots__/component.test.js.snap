// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<Styled(WithStyles(ForwardRef(AppBar)))
  position="static"
>
  <WithStyles(ForwardRef(Container))
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <Connect(Logo) />
      <Connect(withRouter(WithTheme(UserMenu)))
        data-test="userMenu"
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(Container))>
</Styled(WithStyles(ForwardRef(AppBar)))>
`;
